### ISSUE TITLE
Onedrive on system with multiple users

### ISSUE THREAD

**Issue by basd82** [henrypp/simplewall#395]
Hi,

First thanks for this great simple firewall, 

I have one isue i install simplewall for our non profit on our client.
But my isue is that onedrive.exe is on a difrent place for every user is there way to white list i in general undepend where the onedrive.exe is located ?

**Comment by TontyTon** [comment]
As of I know, there is no such option to whitelist onedrive.exe for different users as they have different .exe files.

---

But you can add all the onedrive.exe files and create a single rule to control them.
Settins>Rules>User Rules>Right Click>Add>Select all onedrive.exe files and set the rule as required

**Comment by basd82** [comment]
That's not a Solution When a new user logs in a new Exe file is created
Bas van den Dikkenberg
Op 30 apr. 2019 13:45 schreef Tonty <notifications@github.com>:


As of I know, there is no such option to whitelist onedrive.exe for different users as they have different .exe files.

---

But you can add all the onedrive.exe files and create a single rule to control them.
Settins>Rules>User Rules>Right Click>Add>Select all onedrive.exe files and set the rule as required

**Comment by tokariu** [comment]
simplewall ist just working for the current user. for instance, when you execute a program as another user or start a process with system-user privileges, all applied blocking rules won't work for those and packets go out.
simplewall would need to be working on driver level for that to work. by saying that, I still hope that comes someday.

**Comment by cloudgreen** [comment]
> 
> 
> simplewall ist just working for the current user. for instance, when you execute a program as another user or start a process with system-user privileges, all applied blocking rules won't work for those and packets go out.
> simplewall would need to be working on driver level for that to work. by saying that, I still hope that comes someday.

Oh, this is pretty important information! So all blocking rules are _ignored_ if processes are run as other users or as the system account?

Does Windows Firewall work like that also?

**Comment by baMain** [comment]
@tokariu are you sure about that?
because i see in the Simplewall gui a process named MsMpEng.exe that is running as system on my PC.

@cloudgreen I am 99% sure that Windows Firewall work on all users, and there is high possibility that Simplewall works the same.

both Windows Firewall and Simplewall are based on WFP, you can read more about it here:
https://docs.microsoft.com/en-us/windows/desktop/fwp/windows-filtering-platform-start-page

and by the way there is a similar problem with Program that are updating to a different path(where the path include the app version), this is the case of Windows Defender

**Comment by TontyTon** [comment]
@tokariu 
When you install SW on one user, it works completely for that user. SW is based on WFP. It doesn't care what privileges a process/application trying to connect to network have. It works as configured in SW, even if you open it as another user in the same account.

--
I don't know what happens when there are multiple user accounts.
I think it would work the same, once SW is executed in new account, 'may be' for that you also have to store settings in program folder.

**Comment by henrypp** [comment]
> simplewall ist just working for the current user. for instance, when you execute a program as another user or start a process with system-user privileges, all applied blocking rules won't work for those and packets go out.
simplewall would need to be working on driver level for that to work. by saying that, I still hope that comes someday.

Please let me know, do you enable "Secure filters" option? It can will be related with this option!

**Comment by tokariu** [comment]
@henrypp I don't see this option in my SW version. I'm still using v2.3.13 because i had trouble with the 2.4.x version on win7.

However, I just retried my experiments and from what I see now, SW is working as you might expect, it blocks programs started with system-user privileges.

Let me explain: some SW versions ago, I had to fix some other stuff and needed to execute a program with system-level privilegs. I used "psexec.exe -i -d -s (path to)program.exe" in admin console (https://docs.microsoft.com/en-us/sysinternals/downloads/psexec) to elevate as system user.
That program was completely blocked in SW. But as I started it with system-privileges, it suddenly showed notifications of new versions and new banners, meaning it suddenly had access to the internet. so I came to the logical conclusion that Simplewall works only for the current user, but not for programs that were executed as other users/privileges.

I just retried exactly that with sw v2.3.13 and it correctly blocked the system-privelege program. Obviously something has changed in the meantime. I cannot speak for v2.4.x, though.
